plugins/ecs-bridge/engine/engine.go

```diff
@@ -127,6 +130,21 @@ func (engine *engine) lookupBridge(bridgeName string) (*netlink.Bridge, error) {
 	return bridge, nil
 }
 
+func generateMACAddress() (net.HardwareAddr, error) {
+	buf := make([]byte, 6)
+	var mac net.HardwareAddr
+
+	_, err := rand.Read(buf)
+	if err != nil {
+		return mac, err
+	}
+
+	// Set locally administered addresses bit and reset multicast bit
+	buf[0] = (buf[0] | 0x02) & 0xfe
+	mac = append(mac, buf[0], buf[1], buf[2], buf[3], buf[4], buf[5])
```

Comment:
QQ: Can this MAC address ever collide with any of the other interfaces? Do we need to have a check here?

Reply:
I'm not sure how practical checking this would be at this point. It is very unlikely, I did find this on the internet, for what it's worth (keep in mind this is also a local bridge which maxes out on the order of 1000 interfaces):

MAC addresses are 48 bits long, so there are 281,474,976,710,656 possible MAC addresses. In a network consisting of 1,000,000 (one million) devices, the chance of random MAC address collision is 0,0000000036% (about one in 280,000,000). That's 1000 times less probable than being struck by a lightning.